lxd/instances_post.go

```diff
@@ -404,6 +405,171 @@ func createFromMigration(s *state.State, r *http.Request, projectName string, pr
 	return operations.OperationResponse(op)
 }
 
+// createFromConversion uploads the image to the server's backup directory, converts it into the
+// the raw format, and imports the LXD instance using the converted image.
+func createFromConversion(s *state.State, r *http.Request, projectName string, profiles []api.Profile, req *api.InstancesPost) response.Response {
+	if s.DB.Cluster.LocalNodeIsEvacuated() && r.Context().Value(request.CtxProtocol) != "cluster" {
+		return response.Forbidden(fmt.Errorf("Cluster member is evacuated"))
+	}
+
+	// Validate migration mode.
+	if req.Source.Mode != "push" {
+		return response.NotImplemented(fmt.Errorf("Mode %q not implemented", req.Source.Mode))
+	}
+
+	// Parse the architecture name
+	architecture, err := osarch.ArchitectureId(req.Architecture)
+	if err != nil {
+		return response.BadRequest(err)
+	}
+
+	dbType, err := instancetype.New(string(req.Type))
+	if err != nil {
+		return response.BadRequest(err)
+	}
+
+	// Currently, only VMs can be converted.
+	if dbType != instancetype.VM {
+		return response.BadRequest(fmt.Errorf("Only virtual machines can be created from image conversion. Instance type %q is not supported", req.Type))
+	}
```

Comment:
Actually, we need to support containers, however, no conversion option (e.g. `format`) is allowed in such case.

Reply:
^ To make it equivalent to previous migration. Otherwise, `lxd-migrate` will fail for instances of type container. Alternatively we can fallback to "migration" if instance type is container.